File: Easy/144-binary-tree-preorder-traversal/try-5.cpp
/*
 * Problem #144: Binary Tree Preorder Traversal
 * Difficulty: Easy
 * Submission: Try 5
 * status: Accepted
 * Language: cpp
 * Date: 20/08/2025, 21:50:56
 * Link: https://leetcode.com/problems/binary-tree-preorder-traversal/
 */

/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<int> preorderTraversal(TreeNode* root) {
        stack<TreeNode*> st;
        vector<int> result;
        if(root == NULL){
            return result;
        }

        st.push(root);
        while(!st.empty()){
            TreeNode* ele = st.top();
            st.pop();

            result.push_back(ele->val);
            if(ele->right) st.push(ele->right);
            if(ele->left) st.push(ele->left);

        }
        return result;
    }
};

File: Easy/509-fibonacci-number/try-5.py
/*
 * Problem #509: Fibonacci Number
 * Difficulty: Easy
 * Submission: Try 5
 * status: Accepted
 * Language: python3
 * Date: 13/12/2024, 21:56:02
 * Link: https://leetcode.com/problems/fibonacci-number/
 */

class Solution:
    def fib(self, n: int) -> int:
        c=Counter()
        c[0]=0
        c[1]=1
        for i in range(2,n+1):
            c[i]=c[i-1]+c[i-2]
        return c[n]

    
        
